crates/hyperqueue/src/server/event/log/write.rs

```diff
@@ -1,18 +1,28 @@
 use crate::server::event::log::canonical_header;
-use crate::server::event::MonitoringEvent;
+use crate::server::event::{MonitoringEvent, MonitoringEventId};
 use async_compression::tokio::write::GzipEncoder;
 use async_compression::Level;
+use serde::{Deserialize, Serialize};
 use std::path::Path;
 use tokio::fs::File;
 use tokio::io::AsyncWriteExt;
 
 /// Streams monitoring events into a file on disk.
 pub struct EventLogWriter {
-    file: GzipEncoder<File>,
+    file: File,
     buffer: Vec<u8>,
+
+    /// The ID of the first event in current buffer.
+    first_event_in_buffer: MonitoringEventId,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct EventChunkHeader {
+    pub first_event_id: MonitoringEventId,
+    pub chunk_length: usize,
```

Comment:
We should document what does this length mean (is it compressed/uncompressed size? is it in bytes or event count?). Either by comment (preferred), variable name or type.